### PR TITLE
reduce comparison conditional

### PR DESCRIPTION
**Pull request by guilhermesiani**

### CODE REVIEW

File: src/Computaria/Tardis/Factory.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ public static function grow(
 
     private static function createProxyFactoryIfNecessary(ProxyManager\AbstractBaseFactory $proxyFactory=null)
     {
-        if (false === is_null($proxyFactory)) {
+        if (!is_null($proxyFactory)) {
```
Comment (malukenho) [inline src/Computaria/Tardis/Factory.php:29]: Actually, can be `null !== $proxyFactory`

File: src/Computaria/Tardis/Factory.php [hunk 2]
```diff
@@ -40,7 +40,7 @@ private static function createProxyFactoryIfNecessary(ProxyManager\AbstractBaseF
 
     private static function createIdentityGeneratorIfNecessary(IdentityGenerator $idGenerator=null)
     {
-        if (false === is_null($idGenerator)) {
+        if (!is_null($idGenerator)) {
```
Comment (malukenho) [inline src/Computaria/Tardis/Factory.php:43]: Same as above

Comment (malukenho) [inline src/Computaria/Tardis/Factory.php:43]: Also you can simplify this for:

``` php
return $idGenerator ?: new Identity\MethodCall;
```

File: src/Computaria/Tardis/Factory.php (hunk 1 above)
Comment (guilhermesiani) [inline src/Computaria/Tardis/Factory.php:29]: yes! but do u think that type of condition can be impacted on performance or something else? i did not understand what u mean

Comment (malukenho) [inline src/Computaria/Tardis/Factory.php:29]: @guilhermesiani  Yes. `is_null` is more lazy than `null === $var`. function calls is more lazy then operators usage.

**Comment by augustohp** [comment]
:bow: I really appreciate the contribution, but since it does not affect performance and either functionality I am going to decline this.

As a matter of taste, I prefer the so called [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) with [identical](http://php.net/manual/en/language.operators.comparison.php) operator since they make for less mistakes and are not easy to ignore as an `!`.

PS: I am really sorry for the delay in the response, I saw this early but declined to comment due to not being eager to merge it... I was afraid to discourage you or something. Seems like I've done it without making any comment, so for that I am sorry. Thank you again and keep awesome! :panda_face:

**Pull request closed without merging**